NOTICE

```diff
@@ -3,3 +3,100 @@ Copyright 2023 The Apache Software Foundation
+
+================================================================
+
+This product includes a number of Dependencies with separate copyright notices
+and license terms. Your use of these submodules is subject to the terms and
+conditions of the following licenses.
+
+================================================================
+
+================================================================
+Apache-2.0 licenses
+================================================================
+The following components are provided under the Apache-2.0 License. See project link for details.
+The text of each license is the standard Apache 2.0 license.
+
+* guava (https://github.com/google/guava)
+    Files:
+      java/fury-core/src/main/java/org/apache/fury/util/Preconditions.java
+
+* spark (https://github.com/apache/spark)
+    Files:
+      java/fury-core/src/main/java/org/apache/fury/codegen/Code.java
+      java/fury-core/src/main/java/org/apache/fury/util/Platform.java
+      java/fury-format/src/main/java/org/apache/fury/format/vectorized/ArrowWriter.java
+
+* commons-io (https://github.com/apache/commons-io)
+    Files:
+      java/fury-core/src/main/java/org/apache/fury/io/ClassLoaderObjectInputStream.java
+
+
+* fast-serialization (https://github.com/RuedigerMoeller/fast-serialization)
+    Files:
+      java/fury-benchmark/src/main/java/org/apache/fury/benchmark/Identity2IdMap.java
+
+* ray (https://github.com/ray-project/ray)
+    Files:
+      ci/format.sh
+
+
+================================================================
+BSD-3-Clause licenses
+================================================================
+The following components are provided under the BSD-3-Clause License. See project link for details.
+The text of each license is also included in licenses/LICENSE-[project].txt.
+
+* kryo (https://github.com/EsotericSoftware/kryo)
+    Files:
+      java/fury-benchmark/src/main/java/org/apache/fury/benchmark/data/Data.java
+      java/fury-benchmark/src/main/java/org/apache/fury/benchmark/data/Image.java
+      java/fury-benchmark/src/main/java/org/apache/fury/benchmark/data/Media.java
+      java/fury-benchmark/src/main/java/org/apache/fury/benchmark/data/MediaContent.java
+      java/fury-benchmark/src/main/java/org/apache/fury/benchmark/data/Sample.java
+      java/fury-benchmark/src/main/java/org/apache/fury/benchmark/state/BufferType.java
+      java/fury-benchmark/src/main/java/org/apache/fury/benchmark/state/KryoState.java
+      java/fury-core/src/main/java/org/apache/fury/collection/FuryObjectMap.java
+      java/fury-core/src/main/java/org/apache/fury/collection/IdentityMap.java
+      java/fury-core/src/main/java/org/apache/fury/collection/IdentityObjectIntMap.java
+      java/fury-core/src/main/java/org/apache/fury/collection/LongMap.java
+      java/fury-core/src/main/java/org/apache/fury/collection/ObjectIntMap.java
+      java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java
```

Comment:
It is hard to maintain the list to me. Could we just use wildcard matcher? e.g. `fury-benchmark/*`

Reply:
Fixed.

Reply:
@PragmaTwice I suggest maintaining exact file list, not all of those files comes from kryo

Reply:
For now, I will add the exact file names. In the future, if we feel it's getting hard to maintain, we can replace it with wildcard.

Reply:
Sounds good to me